jvmhs/test/Jvmhs/Analysis/ReduceTest.hs

```diff
@@ -2,40 +2,79 @@
 module Jvmhs.Analysis.ReduceTest where
 
 import SpecHelper
-import Data.Graph.Inductive.Graph
 import Jvmhs
 import Jvmhs.Analysis.Reduce
 
-outputPath :: FilePath
-outputPath = "test/output/interface"
+import qualified Data.Set as S
+import qualified Data.Map as M
 
+--outputPath :: FilePath
+--outputPath = "test/output/interface"
+--
 
-afterReduceInterface :: SpecWith () -> Spec
-afterReduceInterface = before $ reduceInterface classpath outputPath "SimpleI"
+spec_unusedInterfaces :: Spec
+spec_unusedInterfaces =
+  it "should find (unused interfaces, their parents) pairs" $ do
+    Right x <- runTestClassPool $ do
+      (found, _) <- computeClassClosure (S.singleton  ("SimpleI"))
 
-spec_reduce :: Spec
-spec_reduce = afterReduceInterface $ do
-  let getInterfaces className = runTestClassPoolFromPath [outputPath] $ do
-        cls <- loadClass className
-        return (cls ^. classInterfaces)
+      iMap  <- findUnusedInterfaces found
+      return $ M.toList iMap
+    x `shouldMatchList` [("Itfc2", S.fromList ["ItfcParent"])]
 
-  it "after removing interfaces, Itfc2 should replaced by ItfcParent" $ do
-      x <- getInterfaces "SimpleI"
-      x `shouldBe` ["ItfcParent", "Itfc"]
 
+--prop_makeConsistent :: M.Map ClassName [ClassName] -> Property
```

Comment:
This prop has a problem for randomly generated tests:
if we have 
  ["A"] => ["A"]
then we are not able to "inline" anything (infinite loop) if we are trying to satisfy the requirement.
However, this case never happens to the interface hierarchy. So I just commented it and see how can we change the constraint.